### PR TITLE
Fixed integ test to run on all plugins

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Fixed integ test to run on all plugins.
Added UT for the change.
Test Summary:
```
2021-09-28 17:45:39 INFO     PASS: Integration Test for sql with-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for sql without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for alerting without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for k-NN without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for anomaly-detection with-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for anomaly-detection without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for asynchronous-search with-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for asynchronous-search without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for dashboards-reports without-security with status code 0
2021-09-28 17:45:39 INFO     PASS: Integration Test for dashboards-notebooks without-security with status code 0
2021-09-28 17:45:39 ERROR    FAIL: Integration Test for alerting with-security with status code 1
2021-09-28 17:45:39 ERROR    FAIL: Integration Test for index-management with-security with status code 1
2021-09-28 17:45:39 ERROR    FAIL: Integration Test for index-management without-security with status code 1
2021-09-28 17:45:39 ERROR    FAIL: Integration Test for k-NN with-security with status code 1
```
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]

File: bundle-workflow/src/run_integ_test.py [hunk 1]
```diff
@@ -72,11 +73,11 @@ def main():
 
         if passed_components:
             for component, result in passed_components.items():
-                logging.info(f'PASS: Integration Test for {component} {result[1]} with status code {result[0]}')
+                logging.info(f'PASS: Integration Test for {component[0]} {component[1]} with status code {result}')
 
         if failed_components:
             for component, result in failed_components.items():
-                logging.error(f'FAIL: Integration Test for {component} {result[1]} with status code {result[0]}')
+                logging.error(f'FAIL: Integration Test for {component[0]} {component[1]} with status code {result}')
```
Comment (dblock) [inline bundle-workflow/src/run_integ_test.py:80]: Note another problem here that we display results out of order of execution.

Comment (owaiskazi19) [reply bundle-workflow/src/run_integ_test.py:80]: This can be resolve if we use `OrderedDict` instead of `dict`. I can create an issue for the ordering

Comment (dblock) [reply bundle-workflow/src/run_integ_test.py:80]: It should be resolved with a test results class that collects results in a single collection and not two.

Comment (owaiskazi19) [reply bundle-workflow/src/run_integ_test.py:80]: Added the details in #579

Comment (setiah) [reply bundle-workflow/src/run_integ_test.py:80]: +1 to `TestResults` class. It'll make this much cleaner and extensible.

* generateSummaryReport()
* appendResult(component, config, result)

Comment (setiah) [reply bundle-workflow/src/run_integ_test.py:80]: the parsing logic will be replaced by generateSummary method

**Review by setiah — CHANGES_REQUESTED** [review]
https://github.com/opensearch-project/opensearch-build/pull/630/files#r717869688

**Comment by peterzhuamazon** [comment]
> https://github.com/opensearch-project/opensearch-build/pull/630/files#r717869688

Hi @setiah this seems like an enhancement rather than a bug for now.
Can we merge this PR so 1.1.0 tests can be unblock, then @owaiskazi19 can send another PR to improve this order.

Thanks.

**Comment by setiah** [comment]
> > https://github.com/opensearch-project/opensearch-build/pull/630/files#r717869688
> 
> Hi @setiah this seems like an enhancement rather than a bug for now. Can we merge this PR so 1.1.0 tests can be unblock, then @owaiskazi19 can send another PR to improve this order.
> 
> Thanks.

I'm ok with that. @owaiskazi19 can you pls follow up on the comments in a separate PR and lets keep the issue open until then.

**Review by setiah — APPROVED** [review]

**Pull request merged**